storage/remote/write_handler.go

```diff
@@ -115,6 +120,12 @@ func (h *writeHandler) write(ctx context.Context, req *prompb.WriteRequest) (err
 
 	b := labels.NewScratchBuilder(0)
 	var exemplarErr error
+
+	timeLimitApp := &timeLimitAppender{
```

Comment:
One alternative is to create this once on NewWriteHandler and implement it so it's always `maxAheadTime` from now. We would kind of save some lines to understand/read here, plus a few small allocations. Not a blocker.  🤗

Reply:
hmm. in the current design, `timestamp.FromTime(time.Now().Add(maxAheadTime))` is calculated once per request. If it's get initialized `NewWriteHandler`, it needs to be calculated for each condition? Or did I miss something?